test/server/http_time_tracking_filter_integration_test.cc

```diff
@@ -0,0 +1,147 @@
+  void setup(const std::string& config) {
+    config_helper_.addFilter(config);
+    HttpIntegrationTest::initialize();
+  }
+
+  // Fetches a response with request-level configuration set in the request header.
+  Envoy::IntegrationStreamDecoderPtr getResponse(absl::string_view request_level_config,
+                                                 bool setup_for_upstream_request = true) {
+    Envoy::Http::TestRequestHeaderMapImpl request_headers = request_headers_;
+    request_headers.setCopy(Nighthawk::Server::TestServer::HeaderNames::get().TestServerConfig,
+                            request_level_config);
+    return getResponse(request_headers, setup_for_upstream_request);
+  }
+
+  // Fetches a response with the default request headers, expecting the fake upstream to supply
+  // the response.
+  Envoy::IntegrationStreamDecoderPtr getResponse() { return getResponse(request_headers_); }
+
+  // Fetches a response using the provided request headers. When setup_for_upstream_request
+  // is true, the expectation will be that an upstream request will be needed to provide a
+  // response. If it is set to false, the extension is expected to supply the response, and
+  // no upstream request ought to occur.
+  Envoy::IntegrationStreamDecoderPtr
+  getResponse(const Envoy::Http::TestRequestHeaderMapImpl& request_headers,
+              bool setup_for_upstream_request = true) {
+    cleanupUpstreamAndDownstream();
+    codec_client_ = makeHttpConnection(lookupPort("http"));
+    Envoy::IntegrationStreamDecoderPtr response;
+    if (setup_for_upstream_request) {
+      response = sendRequestAndWaitForResponse(request_headers, 0, default_response_headers_, 0);
+    } else {
+      response = codec_client_->makeHeaderOnlyRequest(request_headers);
+      response->waitForEndStream();
+    }
+    return response;
+  }
+
+  const Envoy::Http::TestRequestHeaderMapImpl request_headers_;
+};
+
+class HttpTimeTrackingFilterConfigTest : public testing::Test,
+                                         public Envoy::Event::TestUsingSimulatedTime {};
+
+// WARNING: Don't move this test. Keep it on top.
+// The static stopwatch member of the extension config persists accross tests, and
+// we test the first call to that singleton instance here.
+TEST_F(HttpTimeTrackingFilterConfigTest, GetElapsedNanosSinceLastRequest) {
+  Envoy::Event::SimulatedTimeSystem& time_system = simTime();
+  EXPECT_EQ(Server::HttpTimeTrackingFilterConfig::getElapsedNanosSinceLastRequest(time_system), 0);
+  time_system.setMonotonicTime(1ns);
+  EXPECT_EQ(Server::HttpTimeTrackingFilterConfig::getElapsedNanosSinceLastRequest(time_system), 1);
+}
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, HttpTimeTrackingIntegrationTest,
+                         testing::ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()));
+
+// Verify expectations with static/file-based time-tracking configuration.
+TEST_P(HttpTimeTrackingIntegrationTest, StaticConfiguration) {
```

Comment:
Applies to all tests:  Prefer names that are affirmations of the behavior being tested (usually including a verb)
Something more like:
```
TEST_P(HttpTimeTrackingIntegrationTest, ReturnsPositiveLatencyForStaticConfiguration) {
```